### DISCUSSION TITLE
CSV Importer documentation for POST

### DISCUSSION THREAD

**Discussion by ict** [firefly-iii/firefly-iii#4321] [Firefly III]
The documentation for the CSV importer POST is cut off mid-sentence ("This POST command ---)":
https://docs.firefly-iii.org/csv/usage/post/

I'd like to know if it is possible to submit a CSV to my locally running CSV importer instance by POSTing a CSV as well as a json-configuration.

Thank you for the great effort of creating and maintaining this, I just switched over from YNAB4 using the CSV importer and a mix of custom scripts using the API and everything is incredibly well documented and thought out :)

**Comment by JC5** [comment]
I'll expand the docs, because this isn't very clear documentation indeed.

You need to have the CSV file + the JSON file already available for the importer. Then, you can POST to `https://example.com/autoimport?directory=/some/local/dir` and the importer will import whatever is in that directory.

You can't upload files together with your POST command to be imported "live". But that's a good idea regardless. The current use case supports people who can't trigger the commands from the terminal but can do a post (using curl or something).

Thx for the compliments, I'm glad to hear you like the tools!

**Comment by ict** [comment]
Understood, I can make that work as well, thanks!
But I'd be interested if that feature becomes available.

**Comment by JC5** [comment]
Feature is live in the `develop` tag. Release will probably be next weekend.

- https://hub.docker.com/layers/fireflyiii/csv-importer/develop/images/sha256-56d51bca841b63036a64910573aa8cb0fb3a79a49a2d66a2f64090037882da5d?context=explore
- https://github.com/firefly-iii/csv-importer/tree/develop

**Comment by JC5** [comment]
Documentation is also updated but it'll take some time for the CDN to pick it up.

**Comment by ict** [comment]
Incredible :) Thanks! I'll check it out soon when I have my future auto-importing pipeline figured out.
